content/cpp/concepts/variables/terms/mutable-variables/mutable-variables.md

```diff
@@ -0,0 +1,94 @@
+---
+Title: 'Mutable variables'
+Description: 'In C++, the `mutable` keyword in C++ allows certain class variables to be modified even within const functions, useful for managing internal state flexibly.'
+Subjects:
+  - 'Computer Science'
+  - 'Game Development'
+Tags:
+  - 'Variables'
+CatalogContent:
+  - 'learn-c-plus-plus'
+  - 'paths/computer-science'
+---
+
+In C++, the `mutable` keyword in C++ allows certain class variables to be modified even within const functions, useful for managing internal state flexibly.
+
+## Syntax
+
+To declare a variable as `mutable`, simply place the `mutable` keyword before the variable type:
+
+```pseudo
+mutable type name;
+```
+
+- **type** - The type of the variable (e.g., `int`, `char`).
+- **name** - The name of the variable.
+
+## Example
+
+In the example below, `accessCount` is marked `mutable`, allowing it to be modified within the const `displayData` function:
+
+```cpp
+#include <iostream>
+#include <string>
+
+class Data {
+public:
+    Data(std::string value) : data(value), accessCount(0) {}
+
+    void displayData() const {
+        ++accessCount; // Allowed modification due to `mutable`
+        std::cout << "Data: " << data << ", Access count: " << accessCount << std::endl;
+    }
+
+private:
+    std::string data;
+    mutable int accessCount; // Can be modified in const methods
+};
+
+int main() {
+    Data d("Sample");
+    d.displayData();
+    d.displayData();
+    return 0;
+}
+```
+
+```shell
+Data: Sample, Access count: 1
+Data: Sample, Access count: 2
+```
```

Comment:
Let's add a sentence to introduce the output:

````suggestion
The above code produces the following output:

```shell
Data: Sample, Access count: 1
Data: Sample, Access count: 2
```
````